components/integrations_marketplace/marketplace_modal.jsx

```diff
@@ -0,0 +1,189 @@
+                    return (
+                        <MarketplaceItem
+                            key={p.Manifest.id}
+                            id={p.Manifest.id}
+                            name={p.Manifest.name}
+                            description={p.Manifest.description}
+                            version={p.Manifest.version}
+                            isPrepackaged={false}
+                            downloadUrl={p.DownloadURL}
+                            homepageUrl={p.HomepageURL}
+                            installed={p.InstalledVersion !== ''}
+                            onConfigure={this.close}
+                            onInstalled={this.getMarketplacePlugins}
+                        />);
+                })
+            }
+        </div>);
+    }
+
+    render() {
+        //WIP: To add pagination section
+        const input = (
+            <div className='filter-row filter-row--full'>
+                <div className='col-sm-12'>
+                    <QuickInput
+                        id='searchChannelsTextbox'
+                        ref='filter'
+                        className='form-control filter-textbox search_input'
+                        placeholder={{id: t('marketplace_modal.search'), defaultMessage: 'Search Plugins'}}
+                        inputComponent={LocalizedInput}
+                        onInput={this.doSearch}
+                    />
+                </div>
+            </div>
+        );
+
+        return (
+            <RootPortal>
+                <FullScreenModal
+                    show={Boolean(this.props.show)}
+                    onClose={this.close}
+                >
+                    <div className='modal-marketplace'>
+                        <h1>
+                            <strong>
+                                <FormattedMessage
+                                    id='marketplace_modal.title'
+                                    defaultMessage='Plugins Marketplace'
+                                />
+                            </strong>
+                        </h1>
+                        {input}
+                        <Tabs
+                            id='marketplaceTabs'
+                            className='tabs'
+                            defaultActiveKey='allPlugins'
+                            activeKey={this.state.tabKey}
+                            onSelect={this.handleSelect}
+                            unmountOnExit={true}
+                        >
+                            <Tab
+                                eventKey={MarketplaceTabs.ALL_PLUGINS}
+                                title={localizeMessage('marketplace_modal.tabs.all_plugins', 'All Plugins')}
+                            >
+                                {this.state.loading ?
+                                    <LoadingScreen/> : (
+                                        this.getPluginsListContent(this.props.marketplacePlugins, false)
```

Comment:
This definitely works, but wondering if we can break this down into smaller components. The overall architecture doesn't need to change, this just turns into a more component styled structure.

I was imagining we could do something like:
```
{this.state.loading ? <LoadingScreen/> : (<PluginsTab type={MarketplaceTabs.ALL_PLUGINS />)}

...
{this.state.loading ? <LoadingScreen/> : (<PluginsTab  type={MarketplaceTabs.INSTALLED_PLUGINS />)}
```


```
PluginsTab extends React.Component {
    render() {
        return (
            <div>
                {
                    this.props.list.length == 0 &&
                    <EmptyPlugins
                        text={
                            this.props.type == INSTALLED
                            ? <FormattedMessage
                                id='marketplace_modal.no_plugins_installed'
                                defaultMessage='You do not have any plugins installed.'
                            />)
                            : <FormattedMessage
                                id='marketplace_modal.no_plugins'
                                defaultMessage='There are no plugins available at this time.'
                            />)
                        }
                    >
                    {
                        this.props.type == INSTALLED &&
                        <a onClick={() => this.changeTab(MarketplaceTabs.ALL_PLUGINS)}>
                            <FormattedMessage
                                id='marketplace_modal.install_plugins'
                                defaultMessage='Install Plugins'
                            />
                        </a>
                    }
                    </EmptyPlugins>
                }
                {
                    pluginsArray.map((p) => {
                        return (
                            <MarketplaceItem
                                key={p.Manifest.id}
                                id={p.Manifest.id}
                                name={p.Manifest.name}
                                description={p.Manifest.description}
                                version={p.Manifest.version}
                                isPrepackaged={false}
                                downloadUrl={p.DownloadURL}
                                homepageUrl={p.HomepageURL}
                                installed={p.InstalledVersion !== ''}
                                onConfigure={this.close}
                                onInstalled={this.getMarketplacePlugins}
                            />
                        );
                    })
                }
            </div>
        )
    }
}

EmptyPlugins extends React.Component {
    render() {
        return (
            <div className='no_plugins_div'>
                <br/>
                <PluginIcon className='icon__plugin'/>
                <br/>
                <br/>
                {this.props.text}
                <br/>
                <br/>
                {this.props.children}
            </div>
        );
    }
}
```

Not blocking this review for this refactor, but would be kinda nice if did do it :) 2/5 on this. Using declarative components increases maintainability and reusability.
Thoughts?

Reply:
Agreed that this would be a good improvement. I will wrap up pending error handling case and will come back to this

Reply:
👍